explain-walkthrough.md

```diff
@@ -0,0 +1,206 @@
+
+{{< copyable "sql" >}}
+
+```sql
+SHOW STATS_HEALTHY;
+```
+
+```sql
++-----------+------------+----------------+---------+
+| Db_name   | Table_name | Partition_name | Healthy |
++-----------+------------+----------------+---------+
+| bikeshare | trips      |                |     100 |
++-----------+------------+----------------+---------+
+1 row in set (0.00 sec)
+```
+
+## Identifying optimizations
+
+There are some aspects of the current execution plan that are quite efficient:
+
+* Most of the work is handled inside the TiKV coprocessor. Only 56 rows need to be sent across the network back to TiDB for processing. Each of these rows is short, and contains only the count that matched the selection.
+
+* Both aggregating the count of rows in TiDB (`StreamAgg_20`), and aggregating in TiKV (`└─StreamAgg_9`) are using stream aggregation, which is very efficient in its memory usage.
+
+The biggest issue with the current execution plan, is that the predicate `start_date BETWEEN '2017-07-01 00:00:00' AND '2017-07-01 23:59:59'` does not apply immediately. All rows are read first with a TableFullScan operator, and then a selection is applied afterwards. If we look at the output from `SHOW CREATE TABLE trips` we can see why this is:
+
+{{< copyable "sql" >}}
+
+```sql
+SHOW CREATE TABLE trips\G
+```
+
+```sql
+*************************** 1. row ***************************
+       Table: trips
+Create Table: CREATE TABLE `trips` (
+  `trip_id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `duration` int(11) NOT NULL,
+  `start_date` datetime DEFAULT NULL,
+  `end_date` datetime DEFAULT NULL,
+  `start_station_number` int(11) DEFAULT NULL,
+  `start_station` varchar(255) DEFAULT NULL,
+  `end_station_number` int(11) DEFAULT NULL,
+  `end_station` varchar(255) DEFAULT NULL,
+  `bike_number` varchar(255) DEFAULT NULL,
+  `member_type` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`trip_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin AUTO_INCREMENT=20477318
+1 row in set (0.00 sec)
+```
+
+There is no index on `start_date`! We would need an index in order to push this predicate into an index reader operator. Let's go ahead and add one:
+
+{{< copyable "sql" >}}
+
+```sql
+ALTER TABLE trips ADD INDEX (start_date);
+```
+
+```sql
+Query OK, 0 rows affected (2 min 10.23 sec)
+```
+
+> **Tip:**
+> 
+> The progress of DDL jobs can be monitored with the command [`ADMIN SHOW DDL JOBS`](/sql-statements/sql-statement-admin.md). The defaults in TiDB are carefully chosen so that adding an index does not impact production workloads too much. For testing environments, consider increasing [`tidb_ddl_reorg_batch_size`](/system-variables.md#tidb_ddl_reorg_batch_size) and [`tidb_ddl_reorg_worker_cnt`](/system-variables.md#tidb_ddl_reorg_worker_cnt). On a reference system, a batch size of 10240 and worker count of 32 achieved a 10x improvement over the defaults.
```

Comment:
Does the `defaults` mean batch size and work count?
```suggestion
> You can monitor the progress of DDL jobs using the [`ADMIN SHOW DDL JOBS`](/sql-statements/sql-statement-admin.md) command. The defaults in TiDB are carefully chosen so that adding an index does not impact production workloads too much. For testing environments, consider increasing the [`tidb_ddl_reorg_batch_size`](/system-variables.md#tidb_ddl_reorg_batch_size) and [`tidb_ddl_reorg_worker_cnt`](/system-variables.md#tidb_ddl_reorg_worker_cnt) values. On a reference system, a batch size of 10240 and worker count of 32 can achieve a 10x performance improvement over the defaults.
```

Reply:
Yes